apis/python/src/tiledbsoma/_tdb_handles.py

```diff
@@ -194,16 +243,33 @@ def _opener(
     def schema(self) -> tiledb.ArraySchema:
         return self._handle.schema
 
-    def non_empty_domain(self) -> Tuple[Tuple[int, int], ...]:
-        """
-        Retrieves the non-empty domain for each dimension, namely the smallest
-        and largest indices in each dimension for which the array/dataframe has
-        data occupied.  This is nominally the same as the domain used at
-        creation time, but if for example only a portion of the available domain
-        has actually had data written, this function will return a tighter
-        range.
-        """
-        return self._handle.nonempty_domain()  # type: ignore
+    def non_empty_domain(self) -> Optional[Tuple[Tuple[object, object], ...]]:
+        try:
+            ned = self._handle.nonempty_domain()
+            if ned is None:
+                return None
```

Comment:
Would it make sense to return `()` here and keep this function non-Optional? If there is a reason to return `None` specifically, that is also fine.